### PR TITLE
[noScript] Register

### PR DESCRIPTION
**Pull request by An-dz**
Following #5565 this branch only adds support for **registering** without scripts.

This patch only adds the support but does not automagically make the theme compatible, some changes in the theme must be done. Themes without support still work.

To test this, use my changes in the default theme:
- [nodebb-theme-persona](https://github.com/NodeBB/nodebb-theme-persona/pull/355)

### CODE REVIEW

File: src/controllers/helpers.js [hunk 1]
```diff
@@ -14,6 +14,20 @@ var middleware = require('../middleware');
 
 var helpers = module.exports;
 
+helpers.noScriptErrors = function (req, res, error, httpStatus) {
+	var middleware = require('../middleware');
+	var httpStatusString = httpStatus.toString();
+	middleware.buildHeader(req, res, function () {
+		res.status(httpStatus).render(httpStatusString, {
```
Comment (barisusakli) [inline src/controllers/helpers.js]: res.render only calls renderHeader

**Comment by pitaj** [comment]
@An-dz please address these issues, sorry for the wait.

**Comment by An-dz** [comment]
This one has finished as well, Travis failed but it was not my fault. :grin:

**Comment by julianlam** [comment]
Thank you for your PR @An-dz :smile: Is there one for logins? I did not see one after a short glance through the 3 in core (+composer and persona)

**Comment by An-dz** [comment]
@julianlam Yes, it's #5748.

There's also #5747 for Logout and #5745 for composing.

**Comment by julianlam** [comment]
Thanks @An-dz -- didn't see that one. Ironically, even without merging that one in login worked (though the redirection did not).

**Comment by An-dz** [comment]
That's normal @julianlam, the login patch only adds better error messages and correct redirect.

Logout also only allows the theme to have a `form` with a `button` to do the `POST` and redirect.

**Comment by julianlam** [comment]
GitHub doesn't detect squashed merges (not to mention the re-based parent...), but this has been merged: c4fbed24f82b5ad274705531226056f62fac0062

**Pull request closed without merging**